### PR TITLE
test PR

### PR DESCRIPTION
**Pull request by Andrii-Kozak**

### CODE REVIEW

File: spec/factories/groups.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+FactoryBot.define do
+  # factory :group do
```
Comment (f1nwe) [inline spec/factories/groups.rb:2]: implement valid factories for new models

File: spec/models/group_spec.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+require 'rails_helper'
+
+RSpec.describe Group, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (f1nwe) [inline spec/models/group_spec.rb:4]: ad shoulda-matchers gem, and implements specs for associations

**Review by f1nwe — CHANGES_REQUESTED** [review]

File: app/models/group.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
+class Group < ApplicationRecord
+  has_many :user_groups, dependent: :destroy
+  has_many :users, through: :user_group
+
+  GROUP_TYPE = [MUSIC, FILM, SOCIAL].freeze
```
Comment (pereslop) [inline app/models/group.rb:5]: constants should be defined in the top of file

File: app/models/group.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  enum group_type: GROUP_TYPE
+
```
Comment (pereslop) [inline app/models/group.rb:7]: add some validations, on group name, description, ect...

File: db/migrate/20210115134437_create_groups.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+class CreateGroups < ActiveRecord::Migration[6.1]
+  def change
+    create_table :groups do |t|
+      t.string :group_name
+      t.string :description
+      t.integer :group_type
```
Comment (pereslop) [inline db/migrate/20210115134437_create_groups.rb:6]: just type, instead of group_type

File: app/models/group.rb (hunk 3 above)
Comment (pereslop) [reply app/models/group.rb:5]: read please article https://github.com/rubocop-hq/rails-style-guide#macro-style-methods

**Review by pereslop — APPROVED** [review]

File: db/migrate/20210115134437_create_groups.rb (hunk 5 above)
Comment (Andrii-Kozak) [reply db/migrate/20210115134437_create_groups.rb:6]: done

File: app/models/group.rb (hunk 3 above)
Comment (Andrii-Kozak) [reply app/models/group.rb:5]: done

File: db/migrate/20210115134437_create_groups.rb (hunk 5 above)
Comment (f1nwe) [reply db/migrate/20210115134437_create_groups.rb:6]: don't use "type" names for the columns in models.
This is already reserved for STI.
@pereslop to FYI/

File: spec/factories/groups.rb (hunk 1 above)
Comment (Andrii-Kozak) [reply spec/factories/groups.rb:2]: done

File: spec/models/group_spec.rb [hunk 6]
```diff
@@ -0,0 +1,44 @@
+require 'rails_helper'
+
+RSpec.describe Group, type: :model do
+  let!(:group) { FactoryBot.create(:group, group_name: 'Music', description: 'Oleg Vinnik') }
```
Comment (pereslop) [inline spec/models/group_spec.rb:4]: Do you really need to create group to check validations?build should be enough

**Review by pereslop — APPROVED** [review]

**Pull request merged**